### DISCUSSION TITLE
Does this work also on any non-steam OS?

### DISCUSSION THREAD

**Discussion by dramirezRT** [moraroy/NonSteamLaunchers-On-Steam-Deck#422] [Q&A]
Like PopOS, Ubuntu or so?

**Comment by moraroy** [comment]
@dramirezRT it should, feel free to try! The log will be in your downloads folder!

**Reply by moraroy** [reply]
@dramirezRT let me know if it does or doesn't, I would like to try and make my script work for as much flavors as Linux if possible

**Reply by dramirezRT** [reply]
Will do! I'm building a new pc this week so, when I try it I'll let you know

**Reply by dramirezRT** [reply]
Hey, so it seems it doesn't work right out of the box with Kubuntu (24.04.1 LTS (Noble Numbat)).
Here's a short summary of my process so far:
- zenity needs to be installed `sudo apt install -y zenity`
- Steam needs to be installed (obvious I know, but still :)) `sudo apt install steam`
  - Also needed to manually link `sudo ln -s /usr/games/steam /usr/bin/steam`
  - Proton also needs to be installed
  - It seems the installation for my user was done here: /home/\<user\>/.steam/debian-installation/steamapps/common
  - The runtaime is therefore installed here: /home/\<user\>/.steam/debian-installation/ubuntu12_32/steam-runtime/run.sh

**Reply by dramirezRT** [reply]
I haven't been able to make it fully work. Although with a change on the script to modify the *steam_dir_root* variable by adding the "debian-installation" suffix was able to get the epic games launcher installed here"
`/home/<user>/.local/share/Steam/steamapps/compatdata/NonSteamLaunchers/pfx/drive_c/Program Files (x86)/Epic Games/Launcher/Portal/Binaries/Win32/EpicGamesLauncher.exe`

**Reply by dramirezRT** [reply]
Seems like with this change the decky loader plugin does find the installer and is able to add it to the Non-Steam Apps list

**Reply by moraroy** [reply]
@dramirezRT oh wow this interesting ok, i didnt know that, what i could do is disable zenity entirely for the plugin, since it isnt really needed anymore, would this debian-installation root suffix work for all debian? not just pop?

**Reply by dramirezRT** [reply]
By installer I actually meant the installed App (Epic, GOG, etc)
I am not sure yet for other distros since I just tried Kubuntu, haven't tried any others yet.
Maybe I could try in a VM later this week

**Reply by BartlomiejF** [reply]
I got Battle.net working with ubuntu 22.04. I did "add non-steam game" from steam library, browsed to battlenet laucher.exe, then went to the properties of the launcher and forced compatibility with GE-Proton 9.15. Battle.net launched and I am installing Diablo 4 right now. Will comment if it doesn't work. I would like to move the prefix somewhere else than my home folder in the future but for now I will check if it works.

**Reply by moraroy** [reply]
@BartlomiejF of course it will man

**Reply by giannibat** [reply]
hi, I use Opensuse Aeon, which is immutable and Steam is only installed via flatpak. In my opinion the script only manages a local installation and does not manage the flatpak so the script stops.

sh: line 1: /usr/bin/steam: File or directory does not exist

It would be enough for me to just start the scanner service, because I have already added all the launcher prefixes manually on Steam, with the relative compact data path.
Why not intercept the Steam installation type and try to use a prefix to manage the paths of both the Deck and a generic distro?

[NonSteamLaunchers-install.log](https://github.com/user-attachments/files/17819401/NonSteamLaunchers-install.log)
